src/core_ocean/Registry.xml

```diff
@@ -1949,6 +1979,35 @@
 			description="The effective ocean density within ice shelves based on Archimedes' principle."
 			packages="landIceCouplingPKG"
 		/>
+        <!-- FIELDS FOR GOTM -->
+		<var name="gotmVertViscTopOfCell" type="real" dimensions="nVertLevelsP1 nCells Time" units="m^2 s^{-1}"
+			 description="GOTM: vertical viscosity defined at the cell center (horizontally) and top (vertically)"
+			 packages="forwardMode;analysisMode"
```

Comment:
Here you are adding 3D fields that are allocated whether gotm is in use or not. To fix that, add a new package in Registry.xml called `gotmPKG` and put all of your new variables in that package (not `forwardMode;analysisMode`).  You will need to add code to turn on the package if the first gotm flag is true. Follow the example here:
```
src/core_ocean/driver/mpas_ocn_core_interface.F
238       !
239       ! test for variable bottom drag of momentum, variableBottomDragPKG
240       call mpas_pool_get_package(packagePool, 'variableBottomDragPKGActive', variableBottomDragPKGActive)
241       call mpas_pool_get_config(configPool, 'config_use_variable_drag', config_use_variable_drag)
242       if ( config_use_variable_drag ) then
243         variableBottomDragPKGActive = .true.
244       end if
```
For bfb restarts, add the prognostic variables that are needed for restart to the restart stream in the Registry.xml file here:
```
1550         <stream name="restart"
```
Also, think carefully about which variables go into the state var_struct. Those should really only be the variables we have a prognostic equation for (i.e. use a time-stepping method to advance). All the other variables that are secondary should go into the diagnostic var_struct. 

The difference is important because the state variables have 2 time levels for the timestepping, and the diagnostics only have one timelevel. From Registry:
```
1883     <var_struct name="state" time_levs="2">
...
2220     <var_struct name="diagnostics" time_levs="1">
```
so think carefully about what what variables belong in state. Those get double the memory, and that is needed for the time-stepping routines. The diagnostics are computed from the state variables, but only need a single instance in memory, for one time slice.

Reply:
This is done in 2f63289 and 6693f2b. I think all the GOTM variables here need to be state variables rather than diagnostics variables as values from the previous time step are required to step GOTM forward.